### DISCUSSION TITLE
Inconsistent use of color names across components

### DISCUSSION THREAD

**Discussion by scottanthonymurray** [indiana-university/rivet-source#425] [Feature and enhancement requests]
## Bugs, issues, broken stuff

There are a handful of components which use color to signify something important, but the class name for those colors are inconsistent.  I'll summarize things in a table:

| Color       | Button    | Badge   | Form Validation | Alerts    |
| ---------- | --------- | --------- | ----------------- | --------- |
| Gray         | N/A       | default  | N/A                    | default       |
| Dark Blue | default (referred to as Primary) | action    | N/A                   | N/A        |
| Blue         | N/A       | N/A        | has-info            | info        |
| Green      | success  | success  | is-valid              | success  |
| Yellow      | N/A       | warning | has-warning     | message |
| Red          | danger  | error      | is-invalid           | error       |
| White      | plain      | N/A       | N/A                    | N/A        |

I personally think it would be easiest if there was a standard name for each color along with SASS variables for each of those specific types of color (e.g. `$color-warning: $color-alert-yellow;`).  It would also be great if the components which use those colors referenced the aliases named instead of the raw color variable so it would enforce the consistency.

I would also recommend supporting the same colors on any component that uses color so people can style things using color without having to look up which specific ones are supported for the component they are styling.  This would mean filling in all of the "N/A" cells above.

## Additional Info

This is based off looking through the SASS source and the documentation for the components listed above:

* https://rivet.uits.iu.edu/components/page-content/badges/
* https://rivet.uits.iu.edu/components/forms/buttons/
* https://rivet.uits.iu.edu/components/forms/text-input/
* https://rivet.uits.iu.edu/components/overlays/alerts/

**Comment by zttodd** [comment]
Some improvements have been made to standardize the use of color names, but review needs to be performed to resolve any remaining issues.

See this table of deprecated class names for more information: https://rivet.iu.edu/components/forms/text-input/#deprecated-form-validation-class-names
